webui/react/src/hooks/useMetricNames.ts

```diff
@@ -54,11 +54,15 @@ const useMetricNames = (
           ...newValidationMetrics.map((name) => ({ name, type: MetricType.Validation })),
           ...newTrainingMetrics.map((name) => ({ name, type: MetricType.Training })),
         ];
-        setMetrics((prevMetrics) =>
-          Loadable.getOrElse([], prevMetrics).length === newMetrics.length
-            ? prevMetrics
-            : Loaded(newMetrics),
-        );
+        setMetrics((prevMetrics) => {
+          const previousMetrics = Loadable.getOrElse([], prevMetrics);
+
+          // The metrics have not changed so return the previousm metrics
+          if (previousMetrics.length === newMetrics.length) return prevMetrics;
```

Comment:
@hkang1 I agree that this is the best approach. When I first saw that we were only comparing the length I thought it was odd logic since just comparing the length is not a comprehensive check if either the `newMetrics` or `prevMetrics` are different. 
I was curious if there was a reason we explicitly did not want to merge the results or implement a better `isEqual` check. But I will make this refactor and see if it leads to ant issues.

Reply:
I remember this being a working progress on the streaming end but originally it was only supposed to stream full lists, but once we discovered it didn't I may have forgotten to update this part 😓

Reply:
@hkang1 the comparison logic has been updated.